### ISSUE TITLE
how to make editsource:edit to run before tests

### ISSUE THREAD

**Issue by max8github** [bmc/sbt-editsource#9]
After trying and re-trying , was not able to use this plugin to do token substitution before unit tests are run.

Maybe i am not using sbt well enough, but i was trying something as easy as:
```
test <<= (test in Test) dependsOn (edit in EditSource)
```
to make test do its work after token resolution, but it does not work.
Mayb i am missing something obvious, but it would still be great if this was in the instructions, i think, which i could add myself if so desired.

Please note that when i run `sbt editsource:edit` it does work: files under `target` are resolved.

But when i run `sbt test`, my unit tests fail: if i look into the `target` dir, the files that were correctly resolved by `editsource:edit` are now unresolved (they contain `${basedirectory}` strings inside them).

**Comment by bmc** [comment]
I'll have to look into this one. I never intended this to be used during testing, but it's a reasonable request.

**Comment by max8github** [comment]
Yes, that would be nice. I wish i could help right away, actually, but still have to learn a bit more on how to code sbt plugins and all.

(BTW, if i may ask: if you could share a tentative, rough and uncommitted time on when you could possibly get to it, please let me know: that will help me plan how to get over this problem in the short-medium term)

**Comment by max8github** [comment]
(To give context to this, if you are interested, this is about setting up unit tests with an in-memory database like h2 or hsqldb. In either case, when i create a Connection object in unit tests, the url cannot contain a relative file path: Connection requires it to be absolute. So, if i want to save the url string in `application.conf` under source control, i must use `${baseDirectory}` in it)

**Comment by bmc** [comment]
With the latest version, this _might_ work:

```scala
test in Test := ((test in Test) dependsOn (edit in EditSource)).value
```
